addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -448,6 +457,204 @@ QUnit.test('chatter should become enabled when creation done', async function (a
     );
 });
 
+QUnit.test('Form view not scrolled when switching record', async function (assert) {
+    assert.expect(7);
+
+    this.data['res.partner'].records = [{
+        activity_ids: [],
+        id: 2,
+        display_name: "second partner",
+        description: [...Array(60).keys()].join('\n'),
+        message_ids: [],
+        message_follower_ids: [],
+    }, {
+        activity_ids: [],
+        id: 3,
+        display_name: "third partner",
+        message_ids: [],
+        message_follower_ids: [],
+    }];
+
+    const messages = [...Array(60).keys()].map(id => {
+        let partnerId;
+        if (id % 2 === 0) {
+            partnerId = 2;
+        } else {
+            partnerId = 3;
+        }
+        return {
+            author_id: [10, "Demo User"],
+            body: `<p>Message ${id + 1}</p>`,
+            date: "2019-04-20 10:00:00",
+            id: id + 1,
+            message_type: 'comment',
+            model: 'res.partner',
+            record_name: 'General',
+            res_id: partnerId,
+        };
+    });
+    this.data['mail.message'].records = messages;
+
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+        viewOptions: {
+            currentId: 2,
+            ids: [2, 3]
+        },
+        config: {
+            device: {size_class: 4}
+        }
+    });
+
+    const controllerContentEl = document.querySelector('.o_content');
+    const formViewEl = document.querySelector('.o_form_view');
+    assert.doesNotHaveClass(formViewEl, 'o_xxl_form_view',
+        "Form view is not xxl"
+    );
+
+    assert.strictEqual(
+        document.querySelector('.breadcrumb-item.active').textContent,
+        'second partner',
+        "'second partner' form view is opened"
+    );
```

Comment:
Don't like using breadcrumb content to assert that... but maybe there's nothing much better with view design?

Reply:
It is usually the way we do it. We can check field values though....

Reply:
Yep, unfortunately you can't do much better without changing template of views...
In our components, that's why we use `data-*-local-id`, as a way to have more robust assertion like this.